### PR TITLE
Enrich reviewboard utility class for python

### PR DESCRIPTION
**Pull request by imsut**
add more methods to ReviewBoardServer class to enable users to get changes, diffs, etc.

### CODE REVIEW

**Comment by Yasumoto** [comment]
Mind adding test coverage for the methods you're touching + adding?

**Comment by imsut** [comment]
add unit tests. fix some bugs.
make URL-fragment passed to api_call consistent (no leading '/')

**Comment by imsut** [comment]
ping

**Comment by imsut** [comment]
the failure in Travis CI is irrelevant to this change.

File: tests/python/twitter/common/reviewboard/test_reviewboard.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
+from twitter.common.reviewboard.reviewboard import ReviewBoardServer
+
+import mock
+from mock import call
+import pytest
+
+@mock.patch("twitter.common.reviewboard.reviewboard.ReviewBoardServer.http_request")
```
Comment (Yasumoto) [inline tests/python/twitter/common/reviewboard/test_reviewboard.py]: Mind adding `autospec=True, spec_set=True` to these patches?

File: tests/python/twitter/common/reviewboard/test_reviewboard.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+def test_fetch_review_requests(mock_method):
+  mock_method.return_value = '''{
+    "stat" : "ok",
+    "review_requests" : [
+      { "id" : 123 }, { "id" : 456 }
+    ]
+  }'''
+  rbs = ReviewBoardServer('http://example.com', username='user', password='pass1')
+  assert len(rbs.fetch_review_requests()) == 2
+  mock_method.assert_called_once_with(
```
Comment (Yasumoto) [inline tests/python/twitter/common/reviewboard/test_reviewboard.py]: mind moving this to:

```
assert mock_http_request.mock_calls == [mock.call( "api/review-requests/?start=0&max-results=25",
    None, None, {'Accept': 'application/json'}, method=None)]
```

We've found this helps make mocked calls much more explicit, and prevents typos in the assert calls.

**Comment by Yasumoto** [comment]
Overall LGTM, if you don't mind fixing those two then I'll merge.

Thanks!

**Comment by imsut** [comment]
addressed comments by Yasumoto. thx!

**Comment by Yasumoto** [comment]
Thanks @imsut !

**Pull request merged**